src/main/java/org/example/XMLValidator.java

```diff
@@ -0,0 +1,175 @@
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.Schema;
+import javax.xml.validation.SchemaFactory;
+import javax.xml.validation.Validator;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class XMLValidator {
+    public static void main(String[] args) throws SAXException, IOException, ParserConfigurationException {
+
+        File userSchemaFile = new File("src/main/resources/User.xsd");
+        File userXMLFile = new File("src/main/resources/User.xml");
+        Source userXMLSource = new StreamSource("src/main/resources/User.xml");
+
+        SchemaFactory factory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+        Schema userSchema = factory.newSchema(userSchemaFile);
+        Validator userSchemaValidator = userSchema.newValidator();
+        userSchemaValidator.validate(userXMLSource);
+
+        File reviewSchemaFile = new File("src/main/resources/Review.xsd");
+        File reviewXMLFile = new File("src/main/resources/Review.xml");
+        Source reviewXMLSource = new StreamSource("src/main/resources/Review.xml");
+
+        Schema reviewSchema = factory.newSchema(reviewSchemaFile);
+        Validator reviewSchemaValidator = reviewSchema.newValidator();
+        reviewSchemaValidator.validate(reviewXMLSource);
+
+        File productSchemaFile = new File("src/main/resources/Product.xsd");
+        File productXMLFile = new File("src/main/resources/Product.xml");
+        Source productXMLSource = new StreamSource("src/main/resources/Product.xml");
+
+        Schema productSchema = factory.newSchema(productSchemaFile);
+        Validator productSchemaValidator = productSchema.newValidator();
+        productSchemaValidator.validate(productXMLSource);
+
+        File categorySchemaFile = new File("src/main/resources/Category.xsd");
+        File categoryXMLFile = new File("src/main/resources/Category.xml");
+        Source categoryXMLSource = new StreamSource("src/main/resources/Category.xml");
+
+        Schema categorySchema = factory.newSchema(categorySchemaFile);
+        Validator categorySchemaValidator = categorySchema.newValidator();
+        categorySchemaValidator.validate(categoryXMLSource);
+
+        File addressSchemaFile = new File("src/main/resources/Address.xsd");
+        File addressXMLFile = new File("src/main/resources/Address.xml");
+        Source addressXMLSource = new StreamSource("src/main/resources/Address.xml");
+
+        Schema addressSchema = factory.newSchema(addressSchemaFile);
+        Validator addressSchemaValidator = addressSchema.newValidator();
+        addressSchemaValidator.validate(addressXMLSource);
+
+        DocumentBuilder db = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+        Document userDoc = db.parse(userXMLFile);
+        NodeList userNodeList = userDoc.getElementsByTagName("user");
+
+        System.out.println("Parsing user xml");
```

Comment:
This would probably be better with logger

Reply:
I can make that change as well

Reply:
fixed